queries/cairo/indents.scm

```diff
@@ -0,0 +1,45 @@
+[
+  (mod_item)
+  (struct_item)
+  (enum_item)
+  (impl_item)
+  (struct_expression)
+  (match_expression)
+  (tuple_expression)
+  (match_arm)
+  (match_block)
+  (call_expression)
+  (assignment_expression)
+  (arguments)
+  (block)
+  (use_list)
+  (field_declaration_list)
+  (enum_variant_list)
+  (tuple_pattern)
```

Comment:
Are the items in the list completely non-overlapping?

Example is like #4449, where two captures overlap with one another, leading to doubly-indented patterns

Reply:
Hmm, for the most part I'd say no, Cairo 0.x is python-like and Cairo 1.0 is rust-like, so I mostly just copied the indents that applied, though I'll check some of the potential candidates for double-indents

Reply:
In a Cairo file with just this
```cairo
struct Span<T> {
    snapshot: @Array<T>
}
```

Why would gg=G dedent the field?

Reply:
gg=G only considers the first node of each line, while o considers the last node of previous non blank line as well